papas/pfalgo/pfreconstructor.py

```diff
@@ -179,36 +164,33 @@ def reconstruct_block(self, block):
         for id in ids:
             self.locked[id] = False
         
-        self.debugprint = False
-        if (self.debugprint  and len(block.element_uniqueids)> 4):
-            print  block
-            
        
         if len(ids) == 1: #TODO WARNING!!! LOTS OF MISSING CASES
             id = ids[0]
-            
+            parent_ids = [block.uniqueid, id]
             if Identifier.is_ecal(id):
-                self.insert_particle(block, self.reconstruct_cluster(block.pfevent.ecal_clusters[id],"ecal_in"))
+                self.insert_particle(parent_ids, self.reconstruct_cluster(self.ecals[id], "ecal_in"))
                 
             elif Identifier.is_hcal(id):
-                self.insert_particle(block, self.reconstruct_cluster(block.pfevent.hcal_clusters[id],"hcal_in"))
+                self.insert_particle(parent_ids, self.reconstruct_cluster(self.hcals[id], "hcal_in"))
                 
             elif Identifier.is_track(id):
-                self.insert_particle(block, self.reconstruct_track(block.pfevent.tracks[id]))
+                self.insert_particle(parent_ids, self.reconstruct_track(self.tracks[id]))
                 # ask Colin about energy balance - what happened to the associated clusters that one would expect?
         else: #TODO
             for id in sorted(ids) : #newsort
                 if Identifier.is_hcal(id):
-                    self.reconstruct_hcal(block,id)
+                    self.reconstruct_hcal(block, id)
             for id in sorted(ids) : #newsort
```

Comment:
Left as is. I am not 100% sure but I think we should first reconstruct all hcals before proceeding to tracks etc.  Added comment